### ISSUE TITLE
NoteBlockEvent.Play should use accessors

### ISSUE THREAD

**Issue by SoniEx2** [MinecraftForge/MinecraftForge#2483]
NoteBlockEvent.Play currently uses a `public Instrument instrument` field. This should be changed to accessors and they should be overridable (see eNByeX/NoteBetter#1).

**Comment by LexManos** [comment]
You link to a random repo with nothing on the page but 'Blocked by {link back to this page}' No... That's stupid.
And yes, things should be moved to accessors, but thats a overarching thing that we want to do to potentially move away from hardcoded event classes. But this isn't going to happen for every nitty gritty thing one line at a time.

**Comment by SoniEx2** [comment]
I need this for my mod's event. I need to know any time the setter is called, even when it would be a no-op in the case of a vanilla NoteBlockEvent.Play. This is so instrument overriding works for mods that don't support my custom event.

**Comment by LexManos** [comment]
Be more detailed and explain why.
You shouldn't NEED to know when this is overriden, just at the end of the event there is a new value.

**Comment by SoniEx2** [comment]
If someone sets the instrument to anything, I need that anything to override my custom instrument (different field, different type). Even when the "fake instrument" (the one I originally returned on getInstrument) is the same as the override (the one that's being passed into setInstrument), this should happen.

Currently I can hook it only if the new instrument is different from the fake instrument. This would be useful with e.g. an "instrument selector" block (someone sets the instrument for all iron materials to, say, "note.pling", and then an iron material "instrument selector" block attempts to override it to piano and fails because the event said piano originally).

In other words,

1. Note block is activated.
2. NoteBetter hooks note block.
3. NoteBetter triggers custom event with custom instrument, using "piano" on the forge event's instrument.
4. Block below sets event instrument to "piano".
5. NoteBetter sees piano==piano and plays custom instrument (**BAD UX**)

**Comment by williewillus** [comment]
what part of those 5 steps is forge's problem...exactly? And you have BAD UX on the line with your mod, ironically

**Comment by SoniEx2** [comment]
@williewillus This issue/proposed change is a solution I can apply to my mod. The bad UX applies to both mods, technically.

**Comment by williewillus** [comment]
Make a PR with the changes you want then

lock pls :P
